### PR TITLE
docs(ydb): translate query plan documentation

### LINKED ISSUE #2875: [docs] translate the "query plan optimization" article
https://ydb.tech/docs/ru/operations/query_plans_optimization

### PR DESCRIPTION
**Pull request by ulya-sidorina**

### CODE REVIEW

File: ydb/docs/en/core/yql/query_plans.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+#### Delete
+ Deleting rows from the table.
+
+Attribute | Description
+--- | ---
+Table | table name
+
+#### Join
+Combine two data sources (subqueries or tables), join strategy is specified in the operation description.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:61]:
```suggestion
Combine two data sources (subqueries or tables) by keys. The join strategy is specified in the operation description.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+#### Sort
+Sorting rows.
+
+Attribute | Description
+--- | ---
+SortBy | columns used for sorting
+
+#### TopSort
+Sorting rows with specified limit.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:87]:
```suggestion
Partial rows sorting with a specified limit.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+Attribute | Description
+--- | ---
+TopSortBy | columns used for sorting
+Limit | rows limit
+
+#### Top
+Returns first N elements that are less or equal to the N+1 element if the entire sequence were sorted.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:95]:
```suggestion
Returns the first N elements that are less or equal to the N+1 element if the entire sequence were sorted.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+#### Offset
+Offset, allowing to skip the first N elements of a given set of rows.
+
+Attribute | Description
+--- | ---
+Offset | offset value
+
+#### Union
+Union the results of two or more subqueries into one set.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:117]:
```suggestion
Concatenate the results of two or more subqueries into a single row set.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+#### Iterator
+Iterates through a given set of rows, typically use a [precompute](#precompute) as an argument.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:120]:
```suggestion
Iterates through a given set of rows. Typically uses a [precompute](#precompute) as an argument.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+#### PartitionByKey
+Partitioning by key typically use a [precompute](#precompute).
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:123]:
```suggestion
Partitioning by key. Typically uses a [precompute](#precompute).
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+### Connection
+Data dependencies between stages.
+
+**UI representation**:
+![connection](_assets/connection.png)
+
+Each stage is executed as a certain number of tasks. For example, a reading stage may be executed in N tasks, where N is the number of table shards. Method of transferring data between stages depends on the type of connection. You can find a description of different connections below.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:131]:
```suggestion
Each stage is executed as a certain number of tasks. For example, a reading stage may be executed in N tasks, where N is the number of table shards. The method of transferring data between stages depends on the type of connection. Below is a description of different connections.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+#### UnionAll
+Combines the results of all producer stage tasks and sends them as a single result to a single consumer stage task.
+
+#### Merge
+A special case of `UnionAll`, the results of the producer stage are sorted by a specified set of columns and merged result also is sorted.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:137]:
```suggestion
This is a special case of `UnionAll`. The results of the producer stage are sorted by a specified set of columns and merged into a result that is also sorted.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+#### Broadcast
+Sends the result of a single producer stage task to all consumer stage tasks.
+
+#### Map
+Implements 1-to-1 relationships between tasks of stages, the producer stage and consumer stage should have the same number of tasks.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:143]:
```suggestion
Implements 1-to-1 relationships between tasks of stages, the producer and consumer stages should have the same number of tasks.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+#### HashShuffle
+Sends the results of producer stage tasks to consumer stage tasks based on a certain rule for specified columns. The rule is defined in the code, but the list of columns is specified in connection.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:146]:
```suggestion
Sends the results of producer stage tasks to consumer stage tasks based on a certain rule for specified columns. The rule is defined in the code, but the list of columns is specified in the connection.
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+**UI representation**:
+![resultset](_assets/resultset.png)
+
+### Precompute
+Materialized intermediate result.
+
+**UI representation**:
+![precompute](_assets/precompute.png)
+
+Stages that depend on precomputes should be executed after the precompute have been completed.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:160]:
```suggestion
Stages that depend on precomputes should be executed after the precompute has been completed.
```

File: ydb/docs/en/core/dba/query-plans-optimization.md [hunk 12]
```diff
@@ -1 +1,101 @@
-{% include [no-translation](../_includes/alerts/no-translation.md) %}
\ No newline at end of file
+# Query plans optimization
```
Comment (blinkov) [inline ydb/docs/en/core/dba/query-plans-optimization.md:1]: This branch needs to be merged with `main`, as this folder no longer exists.

**Review by blinkov — DISMISSED** [review]

File: ydb/docs/en/core/yql/query_plans.md [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+--- | ---
+Predicate | filtering condition
+Limit | rows limit
+
+#### Aggregate 
+Grouping rows by the values of the specified columns or expressions.
+
+Attribute | Description
+--- | ---
+GroupBy | columns used for aggregation
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:76]:
```suggestion
GroupBy | columns or expressions used for aggregation
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+Attribute | Description
+--- | ---
+SortBy | columns used for sorting
+
+#### TopSort
+Partial rows sorting with a specified limit.
+
+Attribute | Description
+--- | ---
+TopSortBy | columns used for sorting
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:91]:
```suggestion
SortBy | columns or expressions used for sorting

#### TopSort
Partial rows sorting with a specified limit.

Attribute | Description
--- | ---
TopSortBy | columns or expressions used for sorting
```

File: ydb/docs/en/core/yql/query_plans.md [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+--- | ---
+TopSortBy | columns or expressions used for sorting
+Limit | rows limit
+
+#### Top
+Returns the first N elements that are less or equal to the N+1 element if the entire sequence were sorted.
+
+Attribute | Description
+--- | ---
+TopBy | columns by which the first N rows will be taken
```
Comment (blinkov) [inline ydb/docs/en/core/yql/query_plans.md:99]:
```suggestion
TopBy | columns or expressions by which the first N rows will be taken
```

**Review by fomichev3000 — APPROVED** [review]

**Pull request merged**